threadpoolctl.py

```diff
@@ -205,10 +206,11 @@ def __exit__(self, type, value, traceback):
         self.unregister()
 
     def unregister(self):
-        for lib_controller in self._controller.lib_controllers:
-            # Since we never call get_num_threads after instanciation of
-            # ThreadpoolController, num_threads holds the original value.
-            lib_controller.set_num_threads(lib_controller.num_threads)
+        """Set the limits back to their original values"""
+        for lib_controller, original_info in zip(
+            self._controller.lib_controllers, self._original_info
+        ):
+            lib_controller.set_num_threads(original_info["num_threads"])
 
     def get_original_num_threads(self):
```

Comment:
I wouldn't mind if this would stay public in the context of the previous comment.

Reply:
But no strong opinion either on this one?

Reply:
Let's keep it for backward compat ?